### DISCUSSION TITLE
Fehler beim Kompelieren / Compilation errors

### DISCUSSION THREAD

**Discussion by martyanov85** [AK-Homberger/NMEA2000-Workshop#9] [Q&A]
Hallo AK,
vielen Dank erstmal für diese großartige Arbeit. 
Ich versuche gerade ActisenseListenerSender-ESP32 auf den ESP32-WROOM-32 Dev Board von diymore zu laden, jedoch bekomme ich jedes mal folgenden Fehler beim Kompelieren: 

...............  Documents\Arduino\libraries\NMEA2000_esp32-master\NMEA2000_esp32.cpp: In member function 'void tNMEA2000_esp32::CAN_init()':
.................Documents\Arduino\libraries\NMEA2000_esp32-master\NMEA2000_esp32.cpp:132:9: error: 'gpio_matrix_in' was not declared in this scope
  132 |         gpio_matrix_in(RxPin,CAN_RX_IDX,0);
      |         ^~~~~~~~~~~~~~
................ Documents\Arduino\libraries\NMEA2000_esp32-master\NMEA2000_esp32.cpp:133:9: error: 'gpio_pad_select_gpio' was not declared in this scope; did you mean 'esp_rom_gpio_pad_select_gpio'?
  133 |         gpio_pad_select_gpio(RxPin);
      |         ^~~~~~~~~~~~~~~~~~~~
      |         esp_rom_gpio_pad_select_gpio
................Documents\Arduino\libraries\NMEA2000_esp32-master\NMEA2000_esp32.cpp:200:5: error: 'gpio_matrix_out' was not declared in this scope; did you mean 'gpio_iomux_out'?
  200 |     gpio_matrix_out(TxPin,CAN_TX_IDX,0,0);
      |     ^~~~~~~~~~~~~~~
      |     gpio_iomux_out

exit status 1

Compilation error: exit status 1

Es scheint etwas mit der NMEA2000_esp32.cpp zu tun. Woran könnte das liegen? Die Bibliotheken sind installiert und das Board wird erkannt. Leider komme ich hier nicht weiter und wäre sehr dankbar für die Hilfe.
Schöne Grüße
Slava

**Comment by AK-Homberger** [comment]
Hallo Slava,

hast die Informatinen zur IDE und zur Board-Software-Version gelesen und befolgt? https://github.com/AK-Homberger/NMEA2000-Workshop#vorbereiten-der-arduino-ide

Dei Fehlermeldungen scheinen mit einer nicht kompatiblem Board-Version zusammenzuhängen.

Im Workshop wurde die IDE 1.8.19 und die ESP32 Board-Version 1.0.6 verwendet.

Und wichtig: In der IDE das Board ESP32 Dev Module einstellen.

Poste doch einmal deine Konfiguration.

Viele Grüße
Andreas

**Reply by martyanov85** [reply]
Hallo Andreas,
danke für die schnelle Antwort, Ja, ich habe die Info gelesen habe aber übersehen, dass ich die aktuellste Version von esp32 installiert habe. Daran lag es. Mit der 1.0.6 Version und Board DOIT ESP32 DEVKIT V1 fünktioniert alles. 
Danke noch mal für den tollen Workshop. Ist eine Riesenleistung.

Viele Grüße
Slava
